### PR TITLE
Add an --edit flag to allow editing commit messages

### LINKED ISSUE #26: Edit git commit message
I like to edit my release commit message to include a summary of changes made since the last release. It would be nice if I could pass a `--edit, -e` flag which is forwarded to git commit.

```sh
$ man -P cat git-commit | grep -A 4 '\-\-edit'
       -e, --edit
           The message taken from file with -F, command line with -m, and from
           commit object with -C are usually used as the commit log message
           unmodified. This option lets you further edit the message taken
           from these sources.
```

### PR DESCRIPTION
**Pull request by Avaq**
Closes #26

### CODE REVIEW

File: xyz [hunk 1]
```diff
@@ -65,6 +68,7 @@ dir="$(cd -P "$(dirname "$path")" && pwd)"
 branch=master
 increment=patch
 message_template='Version X.Y.Z'
+commit_flags=''
```
Comment (Avaq) [inline xyz]: I'm not sure if a `commit_flags` string over an `edit_message` boolean is a good idea, but I'm no hero in bash, and the boolean made the `git commit` command look pretty messy.

Comment (davidchambers) [inline xyz]: I think I'd prefer a "Boolean", though I agree it's a bit messy:

```diff
-run "git commit$commit_flags --message '$message'"
+run "git commit$([[ $edit == true ]] && printf ' --edit') --message '$message'"
```

I'd like to preserve the correspondence between command-line arguments and local variables.

File: xyz [hunk 2]
```diff
@@ -86,6 +90,7 @@ while (($# > 0)) ; do
     -b|--branch)    branch="$1"           ; shift ;;
     -i|--increment) increment="$1"        ; shift ;;
     -m|--message)   message_template="$1" ; shift ;;
+    -e|--edit)      commit_flags+=' -e'   ;;
```
Comment (davidchambers) [inline xyz]: For lack of a better order, let's preserve alphabetical order here and in the `--help` text.

**Comment by davidchambers** [comment]
Looks good! I left two comments about minor things.

File: xyz (hunk 2 above)
Comment (Avaq) [inline xyz]: Ah. I ordered it based on my ordering in the `--help` text, which was based on grouping related options together. I guess alphabetical is nicer for being indisputable.

**Comment by Avaq** [comment]
I made the changes, tested with `--dry-run` and observed `--edit` in the `git commit` line, and squashed my commits.

**Comment by davidchambers** [comment]
:deciduous_tree:

**Pull request merged**